File: Offer/2242/main.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func isSubStructure(A *TreeNode, B *TreeNode) bool {
	if A == nil || B == nil {
		return false
	}
	return checkEqual(A, B) || isSubStructure(A.Left, B) || isSubStructure(A.Right, B)
}

func checkEqual(A *TreeNode, B *TreeNode) bool {
	if B == nil {
		return true
	}

	if A == nil || A.Val != B.Val {
		return false
	}

	return checkEqual(A.Left, B.Left) && checkEqual(A.Right, B.Right)
}

File: Offer/2242/main.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def isSubStructure(self, A: TreeNode, B: TreeNode) -> bool:
        if A == None or B == None:
            return False
        return self.checkEqual(A, B) or self.isSubStructure(A.left, B) or self.isSubStructure(A.right, B)

    def checkEqual(self, A:TreeNode, B: TreeNode) -> bool:
        if B == None:
            return True
        if A == None or A.val != B.val:
            return False
        return self.checkEqual(A.left, B.left) and self.checkEqual(A.right, B.right)